File: src/ProjectR/content/scripts/spells/HolyPunch.lua
function GetTargetType()
    return Single
end

function GetName()
return "Holy Punch"
end

function GetDescription()
  return "Punch the enemy with a holy-infused fist."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .2
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { HOL }
end

function SpellEffect()
  local damage = 4.5 * aAD * (aHOL/100) - 1.5 * dDEF
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/HolyPunch.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class HolyPunch : SpellScriptBase
    {
        public override string Name { get { return "Holy Punch"; } }
        public override string Description { get { return "Punch the enemy with a holy-infused fist."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.HOL, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 20; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 4.5 * cAD * (cHOL / 100) - 1.5 * tDEF;
            DealDamage(damage);
        }
    }
}